File: 0278-first-bad-version/0278-first-bad-version.java
/* The isBadVersion API is defined in the parent class VersionControl.
      boolean isBadVersion(int version); */

public class Solution extends VersionControl {
    public int firstBadVersion(int n) {
        int left = 0, right = n;
        
        while(right-left>1){
            int mid = left+(right-left)/2;
            boolean rs = isBadVersion(mid);
                
            if(rs){
                right = mid;
            }else if(!rs){
                left = mid;
            }

        }
        return right;
    }
}

File: 0278-first-bad-version/0278-first-bad-version.py
# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:

class Solution:
    def firstBadVersion(self, n: int) -> int:
        l, r = 1, n
        
        res = 0
        while l <= r:
            mid = (l + r) // 2
            
            flag = isBadVersion(mid)
            if flag:
                res = mid
                r = mid - 1
            else:
                l = mid + 1
        
        return res